### PR TITLE
include the stl headers we use in jdi & emake

### PR DESCRIPTION
**Pull request by fundies**

### CODE REVIEW

File: CompilerSource/JDI/src/API/AST_Export.cpp [hunk 1]
```diff
@@ -22,6 +22,8 @@
 #include "AST.h"
 #include <General/svg_simple.h>
 
+#include <algorithm> // std::max
```
Comment (RobertBColton) [inline CompilerSource/JDI/src/API/AST_Export.cpp]: `<algorithm>` is a C++ header and should go above a C header like `<svg_simple.h>`

File: CompilerSource/JDI/src/Storage/definition.cpp [hunk 2]
```diff
@@ -27,6 +27,8 @@
 #include <cstdio>
 #include <System/builtins.h>
 #include <Parser/handlers/handle_function_impl.h>
+#include <algorithm> // std::max
```
Comment (RobertBColton) [inline CompilerSource/JDI/src/Storage/definition.cpp]: Same here, the C headers should go last. Anything `.h` is a C header.

File: CompilerSource/JDI/src/API/AST_Export.cpp (hunk 1 above)
Comment (fundies) [reply CompilerSource/JDI/src/API/AST_Export.cpp]: nope it should go at the bottom but includes in that file are backwards

File: CompilerSource/JDI/src/Storage/definition.cpp (hunk 2 above)
Comment (fundies) [reply CompilerSource/JDI/src/Storage/definition.cpp]: again your thinking is backwards

Comment (RobertBColton) [reply CompilerSource/JDI/src/Storage/definition.cpp]: C++ headers may include the C headers, not the other way around. Headers likely to be included by other headers go towards the bottom according to Josh/Rusky. Include order goes local headers, distant headers, system/C++ headers, finally followed by C headers.

Comment (fundies) [reply CompilerSource/JDI/src/Storage/definition.cpp]: thanks for proving my case you twat. #include <Parser/handlers/handle_function_impl.h> is a local header and <algorithm> is a system header. the rest of the includes are backwards not mine

Comment (RobertBColton) [reply CompilerSource/JDI/src/Storage/definition.cpp]: Why is it included using the system header syntax then, I wasn't paying attention to the file name. Can Josh please stop abusing C++?

Anyway: https://stackoverflow.com/a/14243688

Comment (fundies) [reply CompilerSource/JDI/src/Storage/definition.cpp]: <> isnt really  system header syntax is just a style choice I applied to the engine but josh never did in jdi. <> and "" do sameshit regarding includes

Comment (RobertBColton) [reply CompilerSource/JDI/src/Storage/definition.cpp]: I'll mark as resolved, but yeah that's why I thought it was a C header. Don't get me wrong, it probably is still a C header, just not a standard one.

**Comment by RobertBColton** [comment]
I don't know about approving changes to JDI.

**Comment by fundies** [comment]
these are simple enough changes even your mom could approve them

**Comment by RobertBColton** [comment]
@fundies He is going to totally cry and I think it's a bad time since he said he is working on some JDI stuff. Even the smallest conflict here will make him cry. However, it's not just that but also that the small changes tend to accumulate rapidly and before you know it you will have changed a lot more than just this.

**Comment by fundies** [comment]
It's always a bad time and he never merges jdi anyway.

**Comment by RobertBColton** [comment]
Well Josh is not a complete loser, like some people I know, and he does have a family and a job. So, the easier you make things for him, the more cooperative he's going to be. We get just as cranky as he does anyway. I think plain additions are not too bad and it's good that they are commented so he would know whether to keep if a conflict did arise.

**Pull request merged**